File: src/dotnet/Trivi/Trivi.WebGui/wwwroot/js/ts/custom/domain/helpers/new-questions/short-answer.ts
import { QuestionType } from "../../enums/question-type";
import { PutShortAnswerApiRequest } from "../../models/question-models";
import { NewQuestion } from "./new-question";


export class NewShortAnswer extends NewQuestion<PutShortAnswerApiRequest>
{
    public readonly getQuestionType = () => QuestionType.ShortAnswer;
    protected readonly _defaultPrompt = "New short answer";

    public toForm()
    {
        const result: PutShortAnswerApiRequest = {
            collectionId: this._collectionId,
            correctAnswer: "None",
            prompt: this._defaultPrompt,
        };

        return result;
    }
}

File: src/dotnet/Trivi/Trivi.Lib/Domain/Models/ShortAnswer.cs
using Trivi.Lib.Domain.Forms;
using Trivi.Lib.Domain.Other;

namespace Trivi.Lib.Domain.Models;

public class ShortAnswer : Question
{
    public override QuestionType QuestionType => QuestionType.ShortAnswer;

    public string? CorrectAnswer { get; set; }


    public static ShortAnswer FromRequestForm(QuestionId questionId, ShortAnswerForm requestForm)
    {
        return new()
        {
            Id = questionId,
            CollectionId = requestForm.CollectionId,
            CorrectAnswer = requestForm.CorrectAnswer,
            Prompt = requestForm.Prompt,
            Points = requestForm.Points,
        };
    }
}